### PR TITLE
➗ Divide raffle and auction bids according to contracts logic

### PR DESCRIPTION
**Pull request by duckception**
Settings: `AUCTION_WINNERS: 10, RAFFLE_WINNERS: 40`
localhost_3000_ (1)

Settings: `AUCTION_WINNERS: 5, RAFFLE_WINNERS: 10`
localhost_3000_

Settings: `AUCTION_WINNERS: 5, RAFFLE_WINNERS: 19`
localhost_3000_ (3)

### CODE REVIEW

File: packages/frontend/src/components/AllBids/AllBidsList.tsx [hunk 1]
```diff
@@ -18,14 +19,19 @@ export const AllBidsList = ({ search }: AllBidsListProps) => {
...
+  const raffleBidsOffset = Math.max(0, bids.length - RAFFLE_PARTICIPANTS_COUNT)
+  const firstRaffleBidIndex =
+    raffleBidsOffset >= AUCTION_PARTICIPANTS_COUNT ? AUCTION_PARTICIPANTS_COUNT : raffleBidsOffset
+
   const auctionBids = useMemo(() => {
-    const sectionBids = bids.length <= AUCTION_PARTICIPANTS_COUNT ? bids : bids.slice(0, AUCTION_PARTICIPANTS_COUNT)
+    const sectionBids = bids.length <= RAFFLE_PARTICIPANTS_COUNT ? [] : bids.slice(0, firstRaffleBidIndex)
     return search ? searchBid(sectionBids) : sectionBids
-  }, [search, bids, searchBid])
+  }, [search, bids, searchBid, firstRaffleBidIndex])
```
Comment (b-tarczynski) [inline packages/frontend/src/components/AllBids/AllBidsList.tsx:29]: Remove `firstRaffleBidIndex` from dependencies like for `raffleBids`

File: packages/frontend/src/components/AllBids/AllBidsList.tsx [hunk 2]
```diff
@@ -18,14 +19,19 @@ export const AllBidsList = ({ search }: AllBidsListProps) => {
     [search]
   )
 
+  const raffleBidsOffset = Math.max(0, bids.length - RAFFLE_PARTICIPANTS_COUNT)
+  const firstRaffleBidIndex =
+    raffleBidsOffset >= AUCTION_PARTICIPANTS_COUNT ? AUCTION_PARTICIPANTS_COUNT : raffleBidsOffset
+
   const auctionBids = useMemo(() => {
-    const sectionBids = bids.length <= AUCTION_PARTICIPANTS_COUNT ? bids : bids.slice(0, AUCTION_PARTICIPANTS_COUNT)
+    const sectionBids = bids.length <= RAFFLE_PARTICIPANTS_COUNT ? [] : bids.slice(0, firstRaffleBidIndex)
```
Comment (b-tarczynski) [inline packages/frontend/src/components/AllBids/AllBidsList.tsx:27]: Can be simplified to `const sectionBids = bids.slice(0, firstRaffleBidIndex)` as in case `bids.length <= RAFFLE_PARTICIPANTS_COUNT` `firstRaffleBidIndex` equals 0.

File: packages/frontend/src/components/AllBids/AllBidsList.tsx [hunk 3]
```diff
@@ -18,14 +19,19 @@ export const AllBidsList = ({ search }: AllBidsListProps) => {
...
     return search ? searchBid(sectionBids) : sectionBids
-  }, [search, bids, searchBid])
+  }, [search, bids, searchBid, firstRaffleBidIndex])
   const raffleBids = useMemo(() => {
-    const sectionBids = bids.length <= AUCTION_PARTICIPANTS_COUNT ? [] : bids.slice(AUCTION_PARTICIPANTS_COUNT)
+    const sectionBids = bids.length > RAFFLE_PARTICIPANTS_COUNT ? bids.slice(firstRaffleBidIndex) : bids
```
Comment (b-tarczynski) [inline packages/frontend/src/components/AllBids/AllBidsList.tsx:31]: Can be simplified to `const sectionBids = bids.slice(0, firstRaffleBidIndex)` as in case `bids.length <= RAFFLE_PARTICIPANTS_COUNT` `firstRaffleBidIndex` equals 0.

File: packages/frontend/src/constants/raffleParticipantsCount.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const RAFFLE_PARTICIPANTS_COUNT = 80
```
Comment (b-tarczynski) [inline packages/frontend/src/constants/raffleParticipantsCount.ts:1]: `RAFFLE_PARTICIPANTS_COUNT` and `AUCTION_PARTICIPANTS_COUNT` should be taken from contracts. Let's do that in separate PR, I added task for that

**Review by b-tarczynski — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/AllBids/AllBidsList.tsx (hunk 3 above)
Comment (duckception) [reply packages/frontend/src/components/AllBids/AllBidsList.tsx:31]: Should be `const sectionBids = bids.slice(firstRaffleBidIndex)` instead of `const sectionBids = bids.slice(0, firstRaffleBidIndex)` but overall you're right. Nice catch 👌

**Review by b-tarczynski — APPROVED** [review]
